pyomo/contrib/community_detection/tests/test_detection.py

```diff
@@ -0,0 +1,321 @@
+                                                             weighted_graph=False, random_seed=5)
+        community_map_c_weighted_with = detect_communities(model, node_type='c', with_objective=True,
+                                                           weighted_graph=True, random_seed=5)
+
+        test_results = (community_map_v_unweighted_without,
+                        community_map_v_weighted_without,
+                        community_map_v_unweighted_with,
+                        community_map_v_weighted_with,
+                        community_map_c_unweighted_without,
+                        community_map_c_weighted_without,
+                        community_map_c_unweighted_with,
+                        community_map_c_weighted_with)
+
+        correct_community_maps = ({0: ['x1', 'x2'], 1: ['x3', 'x4']}, {0: ['x1', 'x2'], 1: ['x3', 'x4']},
+                                  {0: ['x1', 'x2'], 1: ['x3', 'x4']}, {0: ['x1', 'x2'], 1: ['x3', 'x4']},
+                                  {0: ['c1'], 1: ['c2']}, {0: ['c1'], 1: ['c2']}, {0: ['c1', 'obj'], 1: ['c2']},
+                                  {0: ['c1', 'obj'], 1: ['c2']})
+
+        self.assertEqual(correct_community_maps, test_results)
+
+    def test_communities_7(self):
+        model = create_model_6()
+        empty_model = detect_communities(ConcreteModel())
+        bad_model = detect_communities(2)
+        bad_node_type = detect_communities(model, node_type=[])
+        bad_objective = detect_communities(model, with_objective='c')
+        bad_weighted_graph = detect_communities(model, weighted_graph=set())
+        bad_file_destination = detect_communities(model, file_destination=dict())
+        bad_log_level = detect_communities(model, log_level=[])
+        bad_seed_value = detect_communities(model, random_seed='v')
+
+        test_results = (empty_model, bad_model, bad_node_type, bad_objective, bad_weighted_graph, bad_file_destination,
+                        bad_log_level, bad_seed_value)
+
+        correct_community_maps = ({}, None, None, None, None, None, None, None)
+
+
+def create_model_5():  # MINLP written by GAMS Convert at 05/10/19 14:22:56
+    #
+    #  Equation counts
+    #      Total        E        G        L        N        X        C        B
+    #          6        1        0        5        0        0        0        0
+    #
+    #  Variable counts
+    #                   x        b        i      s1s      s2s       sc       si
+    #      Total     cont   binary  integer     sos1     sos2    scont     sint
+    #          7        1        0        6        0        0        0        0
+    #  FX      0        0        0        0        0        0        0        0
+    #
+    #  Nonzero counts
+    #      Total    const       NL      DLL
+    #         37       35        2        0
+    #
+    #  Reformulation has removed 1 variable and 1 equation
```

Comment:
The comments here and above can be removed. You can add a single line saying that it comes from instance XXX.gms in minlplib.com